doc/user-guide/duckarrays.rst

```diff
@@ -3,28 +3,172 @@
+is capable of using multiple different types of arrays does not have to explicitly check that each one it encounters is
+permitted (e.g. `if dask`, `if numpy`, `if sparse` etc.). Instead xarray can take the more permissive approach of simply
+treating the wrapped array as valid, attempting to call the relevant methods (e.g. `.mean()`) and only raising an
+error if a problem occurs (e.g. the method is not found on the wrapped class). This is much more flexible, and allows
+objects and classes from different libraries to work together more easily.
 
 .. note::
 
-    For ``dask`` support see :ref:`dask`.
+    For discussion on exactly which methods a class needs to implement to be considered "numpy-like", see :ref:`internals.duck_arrays`.
+
+Wrapping numpy-like arrays in xarray
+------------------------------------
+
+:py:class:`DataArray` and :py:class:`Dataset` (and :py:class:`Variable`) objects can wrap these numpy-like arrays.
+
+Constructing xarray objects which wrap numpy-like arrays
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+The primary way to create an xarray object which wraps a numpy-like array is to pass that numpy-like array instance directly
+to the constructor of the xarray class. The page on xarray data structures shows how :py:class:`DataArray` and :py:class:`Dataset`
+both accept data in various forms through their ``data`` argument, but in fact this data can also be any wrappable numpy-like array.
+
+For example, we can wrap the sparse array we created earlier inside a new DataArray object:
+
+.. ipython:: python
+
+    s_da = xr.DataArray(s, dims=["i", "j"])
+    s_da
+
+We can see what's inside - the printable representation of our xarray object (the repr) automatically uses the printable
+representation of the underlying wrapped array.
+
+Of course our sparse array object is still there underneath - it's stored under the `.data` attribute of the dataarray:
+
+.. ipython:: python
+
+    s_da.data
+
+Array methods
+~~~~~~~~~~~~~
+
+We saw above that numpy-like arrays provide numpy methods. Xarray automatically uses these when you call the corresponding xarray method:
+
+.. ipython:: python
+
+    s_da.sum(dim="j")
+
+Numpy ufuncs
+~~~~~~~~~~~~
+
+Xarray objects support calling numpy functions direction on the xarray objects, e.g. ``np.func(da)``.
+This also works when wrapping numpy-like arrays:
+
+.. ipython:: python
+
+    np.sum(s_da, axis=1)
+
+Converting wrapped types
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want to change the type inside your xarray object you can use :py:meth:`DataArray.as_numpy`:
+
+.. ipython:: python
+
+    s_da.as_numpy()
+
+This returns a new :py:class:`DataArray` object, but now wrapping a normal numpy array.
+
+If instead you want to convert to numpy and return that numpy array you can use either :py:meth:`DataArray.to_numpy` or
+:py:meth:`DataArray.values` (what is the difference here?).
```

Comment:
`.values` defers to the underlying array type using `np.asarray`. It will fail for `cupy`. `as_numpy` always does the right thing to get back a numpy array.